pkg/agent/agent.go

```diff
@@ -77,17 +75,28 @@ func (a *Agent) run() error {
 		}
 	}
 
-	ctx, err := a.startManager(svid, key, bundle)
+	err = a.startManager(svid, key, bundle)
 	if err != nil {
 		return err
 	}
 
-	a.t.Go(func() error { return a.managerWait(ctx) })
 	a.t.Go(func() error { return a.startEndpoints(bundle) })
+	a.t.Go(a.superviseManager)
+	return nil
+}
 
-	<-a.t.Dying()
+func (a *Agent) superviseManager() (err error) {
+	// Wait until the agent's tomb is dying or the manager stopped working.
+	select {
+	case <-a.t.Dying():
+	case <-a.Manager.Stopped():
+		err = a.Manager.Err()
+		a.mtx.Lock()
+		a.Manager = nil
+		a.mtx.Unlock()
+	}
 	a.shutdown()
```

Comment:
I think not, because when the agent is dying for whatever reason, I want that the manager be shut down also (in this case by calling a.shutdown() which in turn calls a.Manager.Shutdown().